### PR TITLE
allOf Fixes

### PR DESCRIPTION
**Pull request by hzoppetti**

### CODE REVIEW

File: openapi.yaml [hunk 1]
```diff
@@ -503,12 +503,12 @@ paths:
               allOf:
               - $ref: '#/components/schemas/CreditCard'
               - type: object
-              properties:
-                cvv:
-                  type: string
-                  description: >
-                    The Card Verification Value on the back of the card.
-                  example: '123'
+                properties:
```
Comment (nmelehan) [inline openapi.yaml]: hey heather, I think we need to keep the lines below the `- type: object` line in-line with that array item. For example, this is what it looks like in JSON with the indent i inserted:

```{
   "content": {
      "application/json": {
         "schema": {
            "allOf": [
               {
                  "$ref": "#/components/schemas/CreditCard"
               },
               {
                  "type": "object",
                  "properties": {
                     "cvv": {
                        "type": "string",
                        "description": "The Card Verification Value on the back of the card.\n",
                        "example": "123"
                     }
                  }
               }
            ]
         }
      }
   }
}
```

vs. without: 

```{
   "content": {
      "application/json": {
         "schema": {
            "allOf": [
               {
                  "$ref": "#/components/schemas/CreditCard"
               },
               {
                  "type": "object"
               }
            ],
            "properties": {
               "cvv": {
                  "type": "string",
                  "description": "The Card Verification Value on the back of the card.\n",
                  "example": "123"
               }
            }
         }
      }
   }
}
```

In the without indentation scenario, `properties` isn't in the same dictionary with the previous `type:object`

I've done this indenting a few times in this commit

File: openapi.yaml [hunk 2]
```diff
@@ -2416,7 +2416,6 @@ paths:
               - type
               allOf:
               - $ref: '#/components/schemas/DomainRecord'
-              - type: object
```
Comment (nmelehan) [inline openapi.yaml]: I wasn't quite sure what was being gained by adding these `type: objects` in allOfs with only one other $ref. It seems like it's kind of like telling the file to do a union of a set of things (the $ref) and an empty set (`type: object`)--which would just be the set of things?

File: openapi.yaml [hunk 3]
```diff
@@ -8371,7 +8367,7 @@ paths:
                           items:
                             type: object
                             allOf:
-                            $ref: '#/components/schemas/NodeBalancerNode'
+                            - $ref: '#/components/schemas/NodeBalancerNode'
```
Comment (nmelehan) [inline openapi.yaml]: allOf should accept an array, so I think we needed a dash here

File: openapi.yaml [hunk 4]
```diff
@@ -9423,11 +9418,12 @@ paths:
           content:
             application/json:
               schema:
-                type: object
-                properties:
-                  secret_key:
-                    type: string
-                    example: OiA6F5r0niLs3QA2stbyq7mY5VCV7KqOzcmitmHw
+                allOf:
```
Comment (nmelehan) [inline openapi.yaml]: it seems like we do need to have the full schema for an object storage key here. In either case, the response isn't showing the contents of the schema, but that seems like a front-end bug?

File: openapi.yaml (hunk 1 above)
Comment (hzoppetti) [reply openapi.yaml]: Sounds ok to me

File: openapi.yaml (hunk 4 above)
Comment (hzoppetti) [reply openapi.yaml]: Yeah - I think Alban is going t work on this. I'm unable to get this response to show. There are other cases of this as well.

File: openapi.yaml (hunk 3 above)
Comment (hzoppetti) [reply openapi.yaml]: ok

File: openapi.yaml (hunk 2 above)
Comment (hzoppetti) [reply openapi.yaml]: We were only doing it for consistency? If you don't think we need it, then they can be removed. I'm for either case.

**Pull request merged**